### PR TITLE
adding lens distortion effect

### PR DESCRIPTION
**Pull request by danrossi**
Related issue: #440

### Description

Adding initial lens distortion effect and shader. Needs demo to demonstrate value changes.

### CODE REVIEW

File: src/effects/LensDistortionEffect.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+ * Lens distortion effect.
+ * Original shader ported from https://github.com/ycw/three-lens-distortion
+ * @author danielr electroteque@gmail.com
+ */
+export class LensDistortionEffect extends Effect {
+
+	/**
+	 * Constructs a new lens distortion effect.
+	 * @param {Object} [options] - The options.
+	 * @param {BlendFunction} [options.blendFunction=BlendFunction.SCREEN] - The blend function of this effect.
```
Comment (vanruesc) [inline src/effects/LensDistortionEffect.js:16]:
```suggestion
```

File: src/effects/LensDistortionEffect.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
+import { Uniform, Vector2 } from "three";
+import { Effect } from "postprocessing";
+
+import fragmentShader from "./glsl/lens-distortion.frag";
+
+/**
+ * Lens distortion effect.
+ * Original shader ported from https://github.com/ycw/three-lens-distortion
+ * @author danielr electroteque@gmail.com
```
Comment (vanruesc) [inline src/effects/LensDistortionEffect.js:9]:
```suggestion
```

Author info is already included in the git history and shouldn't be maintained in code.

**Review by vanruesc — CHANGES_REQUESTED** [review]
Looks good :+1:

File: src/effects/LensDistortionEffect.js (hunk 1 above)
Comment (danrossi) [reply src/effects/LensDistortionEffect.js]: done

File: src/effects/LensDistortionEffect.js (hunk 2 above)
Comment (danrossi) [reply src/effects/LensDistortionEffect.js]: done

**Comment by vanruesc** [comment]
> So you don't need a demo added ?

No, the demos that are currently public are in maintenance mode and will be removed soon. The manual demos will also be updated again in the near future so I'll add a demo for lens distortion during that process.

**Review by vanruesc — APPROVED** [review]

**Comment by vanruesc** [comment]
Thanks!

**Comment by danrossi** [comment]
Thanks for reworking the shader. I know enough but working out the workflow for this project took a bit. Hopefully it's useful. I am working on XR changes in another branch and updated the demos and manual with XR support and seems to be working. But it sounds like I'll have to redo them. I am unable to generate demos/manual from github pages however. I'll PR that once's it makes it into npm three I guess.

**Comment by vanruesc** [comment]
> I am working on XR changes in another branch and updated the demos and manual with XR support and seems to be working. But it sounds like I'll have to redo them. I am unable to generate demos/manual from github pages however.

I'm not sure I understand what you're trying to do. Why did you change the demos? The manual is a work in progress that will be published with v7. I'ts not supposed to be live yet.

If you'd like to showcase something, consider using codesandbox with custom builds of `three` and `postprocessing` if needed, but please be clear about what you're trying to convey. Otherwise it's just distacting.

> I'll PR that once's it makes it into npm three I guess.

That won't be necessary. XR support is planned to be implemented as part of - or after - v7 and it will be accompanied by a demo. There is no need to add a VR button to every demo.

**Pull request merged**